### PR TITLE
Add category implemented

### PR DESCRIPTION
**Pull request by fedaktaras**
N + 1 not resolved yet

### CODE REVIEW

File: src/test/resources/application.properties [hunk 1]
```diff
@@ -4,3 +4,6 @@ spring.datasource.username=sa
 spring.datasource.password=
 spring.jpa.database-platform=org.hibernate.dialect.H2Dialect
 spring.liquibase.change-log=classpath:/db/changelog/db.changelog-master.yaml
+
+jwt.secret=lskjqlsfjdcsdbgtkedsdfnxdfsdflkhglskglsasfdnauercnawurc
+jwt.expiration=86400000
```
Comment (EduarfF) [inline src/test/resources/application.properties:9]: do env variables

**Review by EduarfF — CHANGES_REQUESTED** [review]

File: src/test/resources/application.properties (hunk 1 above)
Comment (fedaktaras) [reply src/test/resources/application.properties:9]: It can not be built than

File: pom.xml [hunk 2]
```diff
@@ -109,6 +110,21 @@
...
+        <dependency>
+            <groupId>io.jsonwebtoken</groupId>
+            <artifactId>jjwt-impl</artifactId>
+            <version>${jjwt.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>io.jsonwebtoken</groupId>
+            <artifactId>jjwt-jackson</artifactId>
+            <version>${jjwt.version}</version>
+        </dependency>
```
Comment (andrii-hoienko) [inline pom.xml:138]: Looks like in this PR changes from 2 tasks, please prepare PR with relevant changes only

File: src/main/java/com/example/bookstoreapp/controller/BookController.java [hunk 3]
```diff
@@ -70,9 +70,17 @@ public BookDto updateBook(@PathVariable Long id, @RequestBody BookRequestDto boo
...
 
     @ResponseStatus(HttpStatus.NO_CONTENT)
-    @PreAuthorize("hasAnyRole('USER','ADMIN')")
+    @PreAuthorize("hasRole('ADMIN')")
     @DeleteMapping("/{id}")
     public void delete(@PathVariable Long id) {
         bookService.deleteById(id);
     }
+
+    @GetMapping("/category/{categoryId}")
```
Comment (andrii-hoienko) [inline src/main/java/com/example/bookstoreapp/controller/BookController.java]: Looks like wrong name of enpoint
image

File: src/main/java/com/example/bookstoreapp/controller/CategoryController.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Categories management", description = "Endpoints for management categories")
+@RestController
+@RequiredArgsConstructor
+@PreAuthorize("hasRole('ADMIN')")
```
Comment (andrii-hoienko) [inline src/main/java/com/example/bookstoreapp/controller/CategoryController.java]: Why this parameter set on class level? Does it mean that only admin can access categories related endpoint?

File: src/main/java/com/example/bookstoreapp/controller/CategoryController.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void deleteCategory(@PathVariable Long id) {
+        categoryService.deleteById(id);
+    }
+
+    @GetMapping("/{id}/books")
+    @Operation(summary = "Get books by category ID", description = "Get books by category ID")
+    public List<BookDto> getBooksByCategoryId(@PathVariable Long id, Pageable pageable) {
+        return categoryService.getBooksByCategoryId(id, pageable);
+    }
```
Comment (andrii-hoienko) [inline src/main/java/com/example/bookstoreapp/controller/CategoryController.java]: What is the difference between this and 'getBooksByCategory' method in BookController?

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Looks like in this PR changes from 2 tasks, please prepare PR with relevant changes only

File: src/main/java/com/example/bookstoreapp/controller/BookController.java (hunk 3 above)
Comment (fedaktaras) [reply src/main/java/com/example/bookstoreapp/controller/BookController.java]: I have this in CategoryCintroller

File: src/main/java/com/example/bookstoreapp/controller/CategoryController.java (hunk 4 above)
Comment (fedaktaras) [reply src/main/java/com/example/bookstoreapp/controller/CategoryController.java]: My bad.

File: src/main/java/com/example/bookstoreapp/controller/CategoryController.java (hunk 5 above)
Comment (fedaktaras) [reply src/main/java/com/example/bookstoreapp/controller/CategoryController.java]: None. My mistake

File: pom.xml (hunk 2 above)
Comment (fedaktaras) [reply pom.xml:138]: To be honest it is the hardest task for me to keep code clear on branches....

File: src/main/java/com/example/bookstoreapp/config/SecurityConfig.java [hunk 6]
```diff
@@ -30,13 +37,30 @@ public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Excepti
...
                         auth -> auth
-                                .requestMatchers("/auth/**","/swagger-ui/**")
+                                .requestMatchers(
+                                        "/auth/**",
+                                        "/error",
+                                        "/swagger-ui/**",
+                                        "/v3/api-docs/**",
+                                        "/swagger-resources/**",
+                                        "/swagger-ui.html",
+                                        "/webjars/**")
```
Comment (karma-o) [inline src/main/java/com/example/bookstoreapp/config/SecurityConfig.java:47]: changes from previous PRS are still visible here

**Review by karma-o — CHANGES_REQUESTED** [review]
you still have security changes from prev PR visible in here, merge last main into this branch, or rebase this branch onto last main to resolve that problem

File: src/main/java/com/example/bookstoreapp/config/SecurityConfig.java (hunk 6 above)
Comment (fedaktaras) [reply src/main/java/com/example/bookstoreapp/config/SecurityConfig.java:47]: > changes from previous PRS are still visible here

I did rebase.
It took quite long but those changes are still visible here
image

**Review by karma-o — APPROVED** [review]
GJ! Main part is done, but still want to take a closer look into those security changes on QnA

**Pull request merged**